atc/db/wall.go

```diff
@@ -0,0 +1,115 @@
+package db
+
+import (
+	"database/sql"
+	sq "github.com/Masterminds/squirrel"
+	"github.com/concourse/concourse/atc"
+	"time"
+)
+
+//go:generate counterfeiter . Wall
+
+type Wall interface {
+	SetWall(atc.Wall) error
+	GetWall() (atc.Wall, error)
+	Clear() error
+}
+
+type wall struct {
+	conn Conn
+}
+
+func NewWall(conn Conn) Wall {
+	return &wall{conn: conn}
+}
+
+func (w wall) SetWall(wall atc.Wall) error {
+	tx, err := w.conn.Begin()
+	if err != nil {
+		return err
+	}
+
+	defer Rollback(tx)
+
+	_, err = psql.Delete("wall").RunWith(tx).Exec()
+	if err != nil {
+		return err
+	}
+
+	query := psql.Insert("wall").
+		Columns("message")
+
+	if wall.TTL != "" {
+		ttl, err := time.ParseDuration(wall.TTL)
+		if err != nil {
+			return err
+		}
+
+		expiresAt := time.Now().Add(ttl)
+		query = query.Columns("expires_at").Values(wall.Message, expiresAt)
+	} else {
+		query = query.Values(wall.Message)
+	}
+
+	_, err = query.RunWith(tx).Exec()
+	if err != nil {
+		return err
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (w wall) GetWall() (atc.Wall, error) {
+	var wall atc.Wall
+
+	row := psql.Select("message", "expires_at").
+		From("wall").
+		Where(sq.Or{
+			sq.Gt{"expires_at": time.Now()},
+			sq.Eq{"expires_at": nil},
+		}).
+		RunWith(w.conn).QueryRow()
+
+	err := scanWall(&wall, row)
+	if err != nil && err != sql.ErrNoRows {
+		return atc.Wall{}, err
+	}
+
+	return wall, nil
+}
+
+func scanWall(wall *atc.Wall, scan scannable) error {
+	var (
+		message   sql.NullString
+		expiresAt sql.NullTime
+	)
+
+	err := scan.Scan(&message, &expiresAt)
+	if err != nil {
+		return err
+	}
+
+	if message.Valid {
+		wall.Message = message.String
+	}
```

Comment:
There's a not null constraint on the message column, right? I think you should be able to change `message` in this context to just be a string and then avoid this check since it's enforced by postgres?